### PR TITLE
Add PDC downloads from server shared with CRDC BT-381

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: martha/martha_v3.js [hunk 1]
```diff
@@ -391,7 +408,7 @@ function determineDrsType(url) {
             PROTOCOL_PREFIX_DRS,
             AUTH_SKIPPED,
             BOND_PROVIDER_DCF_FENCE,
-            ACCESS_METHOD_TYPE_NONE,
+            [ACCESS_METHOD_TYPE_GCS, ACCESS_METHOD_TYPE_S3],
```
Comment (breilly2) [inline martha/martha_v3.js:423]: Do we want only S3 for now so we don't turn on signed URLs for GCS objects?

File: martha/martha_v3.js [hunk 2]
```diff
@@ -480,10 +497,10 @@ async function retrieveFromServers(params) {
         url,
     } = params;
 
-    const {sendAuth, bondProvider, accessMethodType} = drsType;
+    const {sendAuth, bondProvider, accessMethodTypes} = drsType;
     console.log(
         `DRS URI '${url}' will use auth required '${sendAuth}', bond provider '${bondProvider}', ` +
-        `and access method type '${accessMethodType}'`
+        `and access method types '${accessMethodTypes.join("', '")}'`
```
Comment (breilly2) [inline martha/martha_v3.js]: Nit: unless you really want the space after the comma, you can use `${accessMethodTypes.toString()}`.

**Review by mcovarr — APPROVED** [review]

File: test/martha/martha_v3.test.js [hunk 3]
```diff
@@ -555,6 +557,58 @@ test.serial('martha_v3 parses a Gen3 CRDC CIB URI response correctly', async (t)
     sinon.assert.callCount(getJsonFromApiStub, 2);
 });
 
+test.serial('martha_v3 parses PDC response correctly', async (t) => {
+    const drsUri = 'drs://dg.4DFC:206dfaa6-bcf1-4bc9-b2d0-77179f0f48fc';
+    const {
+        access_methods: { 0: { access_id: accessId, access_url: { url: s3Url } } },
+    } = pdcResponse;
+    const objectId = drsUri.replace(/.*:/, '');
```
Comment (breilly2) [inline test/martha/martha_v3.test.js]: Might be more clear to build `drsUri` from `objectId`:
```
const objectId = '206dfaa6-bcf1-4bc9-b2d0-77179f0f48fc'
const drsUri = `drs://dg.4DFC:${objectId}`
```

**Review by breilly2 — APPROVED** [review]

File: martha/martha_v3.js (hunk 1 above)
Comment (kshakir) [reply martha/martha_v3.js:423]: Discussed. Turns out CRDC hosts data on both and we want to retrieve GCS data if available. Meanwhile, changed BDC's `.accessMethodTypes` to `[ACCESS_METHOD_TYPE_GCS]`

File: martha/martha_v3.js [hunk 4]
```diff
@@ -369,7 +397,7 @@ function determineDrsType(url) {
             PROTOCOL_PREFIX_DRS,
             AUTH_SKIPPED,
             BOND_PROVIDER_ANVIL,
-            ACCESS_METHOD_TYPE_NONE,
+            ACCESS_METHOD_TYPES_NONE,
```
Comment (mcovarr) [inline martha/martha_v3.js]: TOL would `[ACCESS_METHOD_TYPE_GCS]` be more correct here too?

**Review by mcovarr — APPROVED** [review]

**Review by breilly2 — APPROVED** [review]
👍   I've been waiting on non-prod PDC test data, but still haven't gotten any. Ultimately, I ran locally with a tweak to the preference order for CRDC to be S3, GS. I used the CRDC object used in our tests (since that has both GS and S3) and successfully retrieved a signed URL.

**Comment by breilly2** [comment]
Michael Baumann managed to mine some "real" (staging) PDC data. Example: drs://dg.4DFC:dg.4DFC/f2ffba75-5197-11e9-9a07-0a80fada099c. I tested this branch against this and successfully retrieved a signed URL and downloaded the resulting "fasta" file (small, looks to be truncated or synthetic).

**Pull request merged**